src/Service.Tests/Configuration/Telemetry/AzureLogAnalyticsTests.cs

```diff
@@ -0,0 +1,139 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.ConfigurationTests;
+
+namespace Azure.DataApiBuilder.Service.Tests.Configuration.Telemetry;
+
+/// <summary>
+/// Contains tests for Azure Log Analytics functionality.
+/// </summary>
+[TestClass, TestCategory(TestCategory.MSSQL)]
+public class AzureLogAnalyticsTests
+{
+    public TestContext TestContext { get; set; }
+
+    private const string CONFIG_WITH_TELEMETRY = "dab-azure-log-analytics-test-config.json";
+    private const string CONFIG_WITHOUT_TELEMETRY = "dab-no-azure-log-analytics-test-config.json";
+    private static RuntimeConfig _configuration;
+
+    /// <summary>
+    /// This is a helper function that creates runtime config file with specified telemetry options.
+    /// </summary>
+    /// <param name="configFileName">Name of the config file to be created.</param>
+    /// <param name="isTelemetryEnabled">Whether telemetry is enabled or not.</param>
+    /// <param name="telemetryConnectionString">Telemetry connection string.</param>
+    public static void SetUpTelemetryInConfig(string configFileName, bool isLogAnalyticsEnabled, string logAnalyticsCustomTable, string logAnalyticsDcrImmutableId, string logAnalyticsDceEndpoint)
+    {
+        DataSource dataSource = new(DatabaseType.MSSQL,
+            GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+        _configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions: new(), restOptions: new());
+
+        TelemetryOptions _testTelemetryOptions = new(AzureLogAnalytics: new AzureLogAnalyticsOptions(isLogAnalyticsEnabled, new AzureLogAnalyticsAuthOptions(logAnalyticsCustomTable, logAnalyticsDcrImmutableId, logAnalyticsDceEndpoint)));
+        _configuration = _configuration with { Runtime = _configuration.Runtime with { Telemetry = _testTelemetryOptions } };
+
+        File.WriteAllText(configFileName, _configuration.ToJson());
+    }
+
+    /// <summary>
+    /// Cleans up the test environment by deleting the runtime config with telemetry options.
+    /// </summary>
+    [TestCleanup]
+    public void CleanUpTelemetryConfig()
+    {
+        if (File.Exists(CONFIG_WITH_TELEMETRY))
+        {
+            File.Delete(CONFIG_WITH_TELEMETRY);
+        }
+
+        if (File.Exists(CONFIG_WITHOUT_TELEMETRY))
+        {
+            File.Delete(CONFIG_WITHOUT_TELEMETRY);
+        }
+    }
+
+    /// <summary>
+    /// Tests if the logs are flushed correctly when Azure Log Analytics is enabled.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow("Information Test Message", LogLevel.Information)]
+    [DataRow("Trace Test Message", LogLevel.Trace)]
+    [DataRow("Warning Test Message", LogLevel.Warning)]
+    public async void TestAzureLogAnalyticsFlushServiceSucceed(string message, LogLevel logLevel)
```

Comment:
Since you also performed a manual test to ensure logs are flowing correctly into the log analytics workspace, please mention that in the PR description in the section How this was tested. Provide a screenshot of the logs that you saw.